drivers/gpio/gpio_pca_series.c

```diff
@@ -0,0 +1,1727 @@
+
+	struct gpio_pca_series_data *data =
+		CONTAINER_OF(gpio_cb, struct gpio_pca_series_data, gpio_cb);
+
+	k_work_submit(&data->int_work);
+}
+
+#endif /* CONFIG_GPIO_PCA_SERIES_INTERRUPT */
+
+/**
+ * }
+ * gpio_pca_zephyr_gpio_api
+ */
+
+static const struct gpio_driver_api gpio_pca_series_api_funcs_extended = {
+	.pin_configure = gpio_pca_series_pin_configure,
+	.port_get_raw = gpio_pca_series_port_read_extended, /* special version used */
+	.port_set_masked_raw = gpio_pca_series_port_set_masked,
+	.port_set_bits_raw = gpio_pca_series_port_set_bits,
+	.port_clear_bits_raw = gpio_pca_series_port_clear_bits,
+	.port_toggle_bits = gpio_pca_series_port_toggle_bits,
+#ifdef CONFIG_GPIO_PCA_SERIES_INTERRUPT
+	.pin_interrupt_configure = gpio_pca_series_pin_interrupt_configure_extended,
+	.manage_callback = gpio_pca_series_manage_callback,
+#endif
+};
+
+/**
+ * @brief Initialization function of pca_series
+ *
+ * This sets initial input/ output configuration and output states.
+ * The interrupt is configured if this is enabled.
+ *
+ * @param dev Device struct
+ * @return 0 if successful, failed otherwise.
+ */
+static int gpio_pca_series_init(const struct device *dev)
+{
+	const struct gpio_pca_series_config *cfg = dev->config;
+	struct gpio_pca_series_data *data = dev->data;
+	int ret = 0;
+
+	if (!device_is_ready(cfg->i2c.bus)) {
+		LOG_ERR("i2c bus device not found");
+		goto out_bus;
+	}
+#ifdef GPIO_NXP_PCA_SERIES_DEBUG
+# ifdef CONFIG_GPIO_PCA_SERIES_CACHE_ALL
+	gpio_pca_series_cache_test(dev);
+# endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+#endif /* GPIO_NXP_PCA_SERIES_DEBUG */
+
+	/** Set cache to initial state */
+#ifdef CONFIG_GPIO_PCA_SERIES_CACHE_ALL
+	ret = gpio_pca_series_reg_cache_reset(dev);
+#else
+	ret = gpio_pca_series_reg_cache_mini_reset(dev);
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+	if (ret) {
+		LOG_ERR("cache init error %d", ret);
+		goto out;
+	}
+	LOG_DBG("cache init done");
+
+	/** device reset */
+	gpio_pca_series_reset(dev);
+	LOG_DBG("device reset done");
+
+	/** configure interrupt */
+#ifdef CONFIG_GPIO_PCA_SERIES_INTERRUPT
+	/** save dev pointer */
+	data->self = dev;
+
+	/** init work obj */
+	k_work_init(&data->int_work, gpio_pca_series_interrupt_worker_extended);
+
+	/** Interrupt pin connected, enable interrupt */
+	if (cfg->gpio_int.port != NULL) {
+		if (!device_is_ready(cfg->gpio_int.port)) {
+			LOG_ERR("Cannot get pointer to gpio interrupt device");
+			ret = -EINVAL;
+			goto out;
+		}
+
+		ret = gpio_pin_configure_dt(&cfg->gpio_int, GPIO_INPUT);
+		if (ret) {
+			goto out;
+		}
+
+		ret = gpio_pin_interrupt_configure_dt(&cfg->gpio_int, GPIO_INT_EDGE_TO_ACTIVE);
+		if (ret) {
+			goto out;
+		}
+
+		gpio_init_callback(&data->gpio_cb, gpio_pca_series_gpio_int_handler,
+				   BIT(cfg->gpio_int.pin));
+
+		ret = gpio_add_callback(cfg->gpio_int.port, &data->gpio_cb);
+	} else {
+		LOG_WRN("pca interrupt enabled w/o int-gpios configured in dts");
+	}
+#else
+	UNUSED(data);
```

Comment:
For these and other `UNUSED` instances, could we simply not declare the unused variable unless an `#if` condition that would use it is true?

Reply:
Yeah, but that would add a lot of `#ifdef .. #endif` just to rule out an definition. I'd say I prefer the current solution.